cpp/open3d/data/ExtractZIP.cpp

```diff
@@ -0,0 +1,237 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// The MIT License (MIT)
+//
+// Copyright (c) 2018-2021 www.open3d.org
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#include "open3d/data/ExtractZIP.h"
+
+// Reference:
+// https://github.com/madler/zlib/blob/master/contrib/minizip/miniunz.c
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unzip.h>
+
+#include <iostream>
+
+#include "open3d/utility/FileSystem.h"
+#include "open3d/utility/Logging.h"
+
+#ifdef WIN32
+#include <direct.h>
+#include <io.h>
+#else
+#include <unistd.h>
+#endif
+
+#ifdef __APPLE__
+// In darwin and perhaps other BSD variants off_t is a 64 bit value, hence no
+// need for specific 64 bit functions
+#define FOPEN_FUNC(filename, mode) fopen(filename, mode)
+#else
+#define FOPEN_FUNC(filename, mode) fopen64(filename, mode)
+#endif
+
+#define CASESENSITIVITY (0)
+// If required in future, the `WRITEBUFFERSIZE` size can be increased to 16384.
+#define WRITEBUFFERSIZE (8192)
+#define MAXFILENAME (256)
+
+namespace open3d {
+namespace data {
+
+static int ExtractCurrentFile(unzFile uf, const std::string &password) {
+    char filename_inzip[256];
+    char *filename_withoutpath;
+    char *p;
+    int err = UNZ_OK;
+    FILE *fout = nullptr;
+    void *buf;
+    uInt size_buf;
+
+    unz_file_info64 file_info;
+    err = unzGetCurrentFileInfo64(uf, &file_info, filename_inzip,
+                                  sizeof(filename_inzip), nullptr, 0, nullptr,
+                                  0);
+
+    if (err != UNZ_OK) {
+        utility::LogError("Error {} with zipfile in unzGetCurrentFileInfo.",
```

Comment:
This information isn't very helpful. We shall call it error code (since it is an integer), or convert it to string. 

Here's what we did for curl:
```cpp
            utility::LogWarning("Download failed with error code: {}.",
                                curl_easy_strerror(res));
```

(We shouldn't `LogError` in this function anyway, this error shall be printing after `ExtractCurrentFile` returns.)

Reply:
Done.